staging/src/k8s.io/apimachinery/pkg/api/validate/schema.go

```diff
@@ -72,3 +72,14 @@ func Optional[T comparable](_ operation.Context, fldPath *field.Path, value, _ *
 	}
 	return field.ErrorList{field.Required(fldPath, "optional value was not specified")}
 }
+
+// MaxItems verifies that the specified slice is not longer than max items.
+func MaxItems[T any](_ operation.Context, fldPath *field.Path, value, _ []T, max int) field.ErrorList {
+	if value == nil {
```

Comment:
Don't really need this - len() on a nil slice is 0

Reply:
Need to reflect to check the type, PTAL the updated logic.